services/ort-run/src/main/kotlin/IssueService.kt

```diff
@@ -93,6 +103,31 @@ class IssueService(private val db: Database) {
         CountByCategory(severityToCountMap)
     }
 
+    private suspend fun collectIssues(ortRunId: Long, ortResult: OrtResult): Set<Issue> {
+        val ortWorkerIssues = listOf(
+            ortResult.getAnalyzerIssues() to AnalyzerRunDao.ISSUE_WORKER_TYPE,
+            ortResult.getAdvisorIssues() to AdvisorRunDao.ISSUE_WORKER_TYPE,
+            ortResult.getScannerIssues() to ScannerRunDao.ISSUE_WORKER_TYPE
+        ).flatMap { (issuesByIdentifier, workerType) ->
+            collectIssuesFromWorker(issuesByIdentifier, workerType)
+        }
+
+        // Collect issues that are not stored in the OrtResult, like issues from the config, reporter or notifier
+        // workers.
+        val dbIssues = db.dbQuery { IssueDao.createFromQuery(createOrtRunIssuesQuery(ortRunId)) }
```

Comment:
I would assume that most issues are already stored in the database; there should only be a few exceptions (e.g. the ones from the dependency graph). Could you please comment which exceptions these are? (The comment is currently the other way round, which does not answer this question.)

Reply:
> I would assume that most issues are already stored in the database

This is correct

> there should only be a few exceptions (e.g. the ones from the dependency graph). Could you please comment which exceptions these are?

The dependency graph issues are the only one that are currently known to me. I rephrased the comment to reflect this.

Reply:
Good point.
The database precision could indeed cause duplicates to not be found.
The other values should be safe for the comparison (and IMO we better show duplicates than removing issues that should have been shown).